solaris/utils/tile.py

```diff
@@ -600,8 +599,54 @@ def rasterize_gdf(gdf, src_shape, burn_value=1,
             ((geom, burn_value) for geom in gdf.geometry),
             out_shape=src_shape,
             transform=src_transform
-            )
+        )
     else:
         img = np.zeros(src_shape).astype(np.uint8)
 
     return img
+
+
+def vector_gdf_get_projection_unit(vector_file):
+    """Get the projection unit for a vector_file or gdf.
+
+    Arguments
+    ---------
+    vector_file : :py:class:`geopandas.GeoDataFrame` or geojson/shapefile
+        A vector file or gdf with georeferencing
+
+    Returns
+    -------
+    unit : String
+        The unit i.e. meters or degrees, of the projection
+    """
+    c = gpd.read_file(vector_file)
```

Comment:
I think this is going to throw an error if `vector_file` is a gdf that's already loaded in (as the arguments docstring suggests it can be). Rather than using `gpd.read_file`, import `_check_gdf_load` from `sol.utils.core` and call that function instead - it checks whether the gdf is already loaded, and if it isn't, loads it in.